### PR TITLE
Convert to TypeScript and add Glint

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Converts everything to TypeScript, adds Glint support, updates C.I. to test all of that.

For review, it'd be best to go through commit-by-commit.

Most of the diff is the lockfile: 
image

### CODE REVIEW

**Comment by nicolechung** [comment]
Screen Shot 2022-10-14 at 8 51 19 AM

I'm getting another error trying to run `pnpm start` from the root.

This is after `rm -rf node_modules`. Going to try clearing my pnpm store/v3.

**Comment by nicolechung** [comment]
> Screen Shot 2022-10-14 at 8 51 19 AM
> 

> I'm getting another error trying to run `pnpm start` from the root.
> 
> This is after `rm -rf node_modules`. Going to try clearing my pnpm store/v3.

Nope, `rm -rf /Users/me/Library/pnpm/store/v3` also didn't fix the `start` error. 

I see ember-data in the lockfile, but not any of the package.json files?

Screen Shot 2022-10-14 at 8 56 33 AM

**Comment by nicolechung** [comment]
How is CI passing but I can't build locally, lol?

**Comment by NullVoxPopuli** [comment]
> How is CI passing but I can't build locally, lol?

Is your latest commit the same as the latest commit on the PR? I removed all the ember-data stuff

**Comment by NullVoxPopuli** [comment]
Can you try nom-bomming? I have this working on 3 machines 😅

**Comment by nicolechung** [comment]
> Can you try nom-bomming? I have this working on 3 machines 😅 (though, this doesn't mean these 3 machines aren't coincidentally in an accidentally working state)

Okay (Friday brain) I figured out it was because I was `rm -rf node_modules` but not the node_modules of each package (test-app, ember-velcro). Works now.

File: test-app/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   },
   "scripts": {
     "build": "ember build --environment=production",
+    "types:watch": "glint --build --watch",
     "lint": "npm-run-all --aggregate-output --continue-on-error --parallel \"lint:!(fix)\"",
     "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:*:fix",
     "lint:hbs": "ember-template-lint .",
```
Comment (nicolechung) [inline test-app/package.json:18]: Suggestion: maybe we should have a `clean` command that removes ALL the node_modules

Comment (NullVoxPopuli) [reply test-app/package.json:18]: do you know how to do that cross-platform? (including windows?)
 
 I wonder if rimraf supports this 🤔

Comment (NullVoxPopuli) [reply test-app/package.json:18]: buuut, I also want to (if I have time soon) -- iterate with @mansona on [bottled-ember](https://github.com/empress/bottled-ember), and see how much of a glint-based app can be inlined without per-project boilerplate. The only thing a test-app really needs is the `glint-registry.d.ts`, and the tests

**Review by nicolechung — APPROVED** [review]
Ran the test app, tests are passing for me. 

Also the types are looking good so far (limited by floating ui, I imagine):
Screen Shot 2022-10-14 at 12 38 53 PM

File: test-app/package.json (hunk 1 above)
Comment (nicolechung) [reply test-app/package.json:18]: > do you know how to do that cross-platform? (including windows?)
> 
> I wonder if rimraf supports this 🤔

True...I've seen it done with rimraf before...then the devs picking up the project were totally confused about why there is a package just to `rm -rf`

Comment (NullVoxPopuli) [reply test-app/package.json:18]: yeah, we should ask microsoft why they don't add `rm` to windows

**Review by camskene — APPROVED** [review]
Yay!

**Pull request merged**